### DISCUSSION TITLE
PDF files (ubuntu claiming text file unable to render)

### DISCUSSION THREAD

**Discussion by minhadona** [git-lfs/git-lfs#5818] [General]
I pushed many large pdfs file using git-lfs. OK, but I clone them, (another machine) and try to open:

`Unable to open document "path.pdf".`
`File type plain text document (text/plain) is not supported`

Yes, _I have git lfs installed_ on the machine I'm trying to open the file as well. 

git lfs version: **git-lfs/3.5.1** (GitHub; linux amd64; go 1.21.7; git e237bb3a)
OS: Ubuntu **22.04.4 LTS**

The pushing machine was also Unix: macOS. The pulling machine, as told, Ubuntu.

help
is all I need :(

ref of tryings: https://stackoverflow.com/questions/73633552/i-pushed-a-large-pdf-file-using-git-lfs-but-have-problem-pulling-it/78698782#78698782

**Comment by chrisd8088** [comment]
You might want to try checking whether the files in your local clone's working tree that are Git LFS pointers have been replaced with objects after the clone.

In your new clone, `git lfs ls-files` will list any blobs managed by Git LFS.  If you see `*` in the listing for the file in question, the Git LFS pointer has been replaced in your working tree with what Git LFS understands to be the appropriate full file.  Take a look at the file's size -- it should be identical to what you expect.  If it's not the right size, nor opening as a PDF, that suggests it wasn't properly stored in Git LFS in the first place.

If you see `-`, then you only have a Git LFS pointer file.  You can ask Git LFS to replace it with the corresponding full object file by running `git lfs checkout`.  Then `git lfs ls-files` should report `*` for the file and it should have the expected contents.

As well as using these commands, I'd check some basic things.  What size is your `path.pdf` file?  What does `cat path.pdf` produce?  Does it look like a small Git LFS pointer file, like this?  If so, then `git lfs checkout` should work to replace it in your working tree with the corresponding full file.
```
version https://git-lfs.github.com/spec/v1
oid sha256:aec070645fe53ee3b3763059376134f058cc337247c978add178b6ccdfb0019f
size 7
```

You can also check what Git has in its repository for the file.  If the file is being tracked as a Git LFS object, `git show HEAD:path.pdf` should show a small pointer file like the one above.  If it doesn't show that, then it's not being tracked by Git LFS at all.

**Reply by minhadona** [reply]
I'm trying to absorb the first instructions to follow then, but meanwhile: they are all 131 bytes after using LFS. 
The original size? No idea, I don't have the originals anymore (and that's what's worrying me) 

I don't think there's something linked to local clone because this happens on another Ubuntu too (that is actually in 20 LTS version I guess). I'll check on Windows as soon as I have the opportunity (11th or 12th July I'll be back to tell if some file opened on windows 10)

**Reply by minhadona** [reply]
But yes, **cat**ting 3 of the files produce:

```
version https://git-lfs.github.com/spec/v1
oid sha256:01c2d46888faa69aba78510945f6595abb57c8b74c32ef416a596dc046b16ec5
size 248982
```

```
version https://git-lfs.github.com/spec/v1
oid sha256:73c033801fb07fe60330882ceffaeaad1110aae21713109d142dc3aa9a99b512
size 233758
```

```
version https://git-lfs.github.com/spec/v1
oid sha256:e47c56bfa5e9405382c07e9f63656b3ef27c6fcbf60827319fbf676fe2f207ba
size 221073
```

What have I done, Jesus 
I'll study the instructions, and drop here if this solves on one of the three machines, thanks a lot

**Reply by chrisd8088** [reply]
OK, so that sounds like you've converted files into Git LFS objects and pointers.  You said you pushed the files after performing the conversion, so if that push was successful, then the remote host should have the Git LFS objects which correspond to the pointers.  Using `git lfs pull` and `git lfs checkout` should help in this case.

If you're stuck, then the machine where you performed the conversion should also have copies in its local cache, assuming you haven't deleted the original repository.  You can check in its `.git/lfs/objects` cache, and also its working tree.  If you find the files there, make sure you've successfully pushed them to your remote, and then your clones may work better.

If you pushed to a commercial service, like GitHub, you can always reach out to the service's support team for help.
